### DISCUSSION TITLE
Initialize Node with Reactive Input

### DISCUSSION THREAD

**Discussion by doc987** [mobxjs/mobx-state-tree#2036] [Q&A]
Suppose there is a `ReusableNode` that has a `computed` value which needs input that is external to `self`.  If the input was fixed, then a new property could be added to the `ReusableNode` model, and the fixed value passed in on creation.  What if the input needs to be reactive?  For example, in the code below, what if the `Root` node wanted to give a property (e.g. `root.a`), or give a computed value (e.g. `root.fn`), as the external input value that `root.n.computed` would use?  How is that done?  Alternatively, maybe I'm thinking about this wrong, and there's a better approach.

```js
import * as mst from "./mobx-state-tree.module.js";

const ReusableNode = mst.types
	.model({
		x: 0,
	})
	.views( self => ({
		get computed(){
			return /* something that depends on self.x and something external to this node */;
		}
	}))
;

const Root = mst.types
	.model({
		a: 1,
		b: 1,
		n: mst.types.optional(ReusableNode,{}), // How to provide reactive value in initialization?
	})
	.views( self => ({
		get fn(){ return self.a + self.b; },
	}))
;

const root = Root.create();
```

**Comment by coolsoftwaretyler** [comment]
Hey @doc987 - sorry it took so long for folks to get back to you here. I know it's been almost three years, so if you've since moved on from your MST question, I totally understand.

I'm not 100% sure that this is the answer, but I think what you're probably looking for is something like MST's dependency injection with `getEnv`: https://mobx-state-tree.js.org/concepts/dependency-injection

Would something like that solve this use case? Did you stumble upon that after writing this issue? Find another workaround? 

Let me know! Happy to chat about it some more if it's still interesting to you all these years later. If I don't hear back, I'll probably close this out in two weeks or so.

**Comment by coolsoftwaretyler** [comment]
Oh y'know, I'm actually going to convert this over to a discussion, which will close out the issue, but create a new discussion thread where we can chat more, and/or you can mark my answer (or someone else's) as correct.

Thanks!
